src/proj7BittingCerratoCohenEllmer/bantam/lexer/Scanner.java

```diff
@@ -248,25 +283,27 @@ private boolean isCompleteString(String spelling) {
      * @return returns true if an identifier
      */
     private boolean isCompleteIdentifier(String spelling) {
-        //must start with a letter
-        char start = spelling.charAt(0);
-        if(start >= 'A' && start <= 'Z' ||
-                start >= 'a' && start <= 'z'){
-            // only contains letters, numbers, and underscores
-            for (char character : spelling.toCharArray()) {
-                if(character < '0' ||
-                        character > '9' && character < 'A' ||
-                        character > 'Z' && character < '_' ||
-                        character > '_' && character < 'a' ||
-                        character > 'z'){
-                    errorHandler.register(Error.Kind.LEX_ERROR, sourceFile.getFilename(),
-                            sourceFile.getCurrentLineNumber(),
-                            "Unsupported Symbol, " + character + " in: " + spelling);
-                    return false;
+        if(endOfToken){
```

Comment:
After a bit of reading through this code, I'm fairly sure that it would work but it relies on the monstrously ugly scan method in this branch and can be an extremely small, simple, and elegant method once the stack refactor is complete. It will be a simple matter of checking the top element to see if it is alphabetical.